pkgs/applications/networking/instant-messengers/element/element-web.nix

```diff
@@ -18,73 +18,91 @@ let
+      # see https://medium.com/the-node-js-collection/node-js-17-is-here-8dba1e14e382#5f07
+      # and https://github.com/vector-im/element-web/issues/21043
+      export NODE_OPTIONS=--openssl-legacy-provider
+      mkdir -p $HOME
+
+      fixup_yarn_lock yarn.lock
+      yarn config --offline set yarn-offline-mirror $offlineCache
+      yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules
+
+      runHook postConfigure
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+
+      export VERSION=${version}
+      yarn build:res --offline
+      yarn build:module_system --offline
+      yarn build:bundle --offline
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      cp -R webapp $out
+      cp ${jitsi-meet}/libs/external_api.min.js $out/jitsi_external_api.min.js
+      echo "${version}" > "$out/version"
+      jq -s '.[0] * $conf' "config.sample.json" --argjson "conf" '${builtins.toJSON noPhoningHome}' > "$out/config.json"
+
+      runHook postInstall
+    '';
+
+    meta = {
+      description = "A glossy Matrix collaboration client for the web";
+      homepage = "https://element.io/";
+      changelog = "https://github.com/vector-im/element-web/blob/v${version}/CHANGELOG.md";
+      maintainers = lib.teams.matrix.members;
+      license = lib.licenses.asl20;
+      platforms = lib.platforms.all;
+    };
   };
+in
+if (conf == { }) then unwrapped else
+stdenv.mkDerivation rec {
+  pname = "${unwrapped.pname}-wrapped";
+  inherit (unwrapped) version src meta;
 
-  nativeBuildInputs = [ yarn fixup_yarn_lock jq nodejs ];
-
-  configurePhase = ''
-    runHook preConfigure
-
-    export HOME=$PWD/tmp
-    # with the update of openssl3, some key ciphers are not supported anymore
-    # this flag will allow those codecs again as a workaround
-    # see https://medium.com/the-node-js-collection/node-js-17-is-here-8dba1e14e382#5f07
-    # and https://github.com/vector-im/element-web/issues/21043
-    export NODE_OPTIONS=--openssl-legacy-provider
-    mkdir -p $HOME
-
-    fixup_yarn_lock yarn.lock
-    yarn config --offline set yarn-offline-mirror $offlineCache
-    yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
-    patchShebangs node_modules
-
-    runHook postConfigure
-  '';
-
-  buildPhase = ''
-    runHook preBuild
-
-    export VERSION=${version}
-    yarn build:res --offline
-    yarn build:module_system --offline
-    yarn build:bundle --offline
-
-    runHook postBuild
-  '';
+  nativeBuildInputs = [ jq ];
 
   installPhase = ''
     runHook preInstall
 
-    cp -R webapp $out
-    cp ${jitsi-meet}/libs/external_api.min.js $out/jitsi_external_api.min.js
-    echo "${version}" > "$out/version"
-    jq -s '.[0] * .[1]' "config.sample.json" "${configOverrides}" > "$out/config.json"
+    mkdir -p $out
+    ln -s ${unwrapped}/* $out
```

Comment:
This should probably use lndir

Reply:
What's the benefit? Firefox uses ln directly so I followed it.

Reply:
If you want to recursively create a symlink tree then lndir is nice.